magellan/sampler/down_sample.py

```diff
@@ -0,0 +1,235 @@
+    for x in cols:
+        col_list.append(table.columns.get_loc(x))
+
+    return col_list
+
+
+# create inverted index from token to position
+def _inv_index(table):
+    """
+    This is inverted index function that builds inverted index of tokens on a table
+
+    Args:
+        table: dataframe <input table>
+    
+    Returns:
+        inverted index of type dictionary
+
+    Example:
+        inv_index = _inv_index(table)
+    
+    """
+
+    stop_words = set(_get_stop_words())
+    str_cols_ix = _get_str_cols_list(table)
+    n = len(table)
+    key_pos = dict(zip(range(n), range(n)))
+    inv_index = dict()
+    pos = 0
+    for row in table.itertuples():
+        s = ''
+        for ix in str_cols_ix:
+            s += str(row[ix+1]).lower() + ' '
+        s = s.rstrip()
+        # tokenize them
+        s = set(s.split())
+        s = s.difference(stop_words)
+        for token in s:
+            lst = inv_index.get(token, None)
+            if lst is None:
+                inv_index[token] = [pos]
+            else:
+                lst.append(pos)
+                inv_index[token] = lst
+        pos += 1
+    return inv_index
+
+
+def _probe_index(b_table, y, s_tbl_sz, s_inv_index):
+    """
+    This is probe index function that probes the second table into inverted index to get good coverage in the down sampled output
+
+    Args:
+        b_table: dataframe <input table B>
+        s_tbl_sz: size of table A
+        y: down_sampled size of table A should be close to size * y
+        s_inv_index: inverted index built on Table A
+    
+    Returns:
+        set with indexes
+    
+    """
+    y_pos = math.floor(y/2)
+    h_table = set()
+    stop_words = set(_get_stop_words())
+    str_cols_ix = _get_str_cols_list(b_table)
+    
+    # Progress Bar
+    bar = pyprind.ProgBar(len(b_table))
+
+    for row in b_table.itertuples():
+        bar.update()
+        s = ''
+        for ix in str_cols_ix:
+            s += str(row[ix+1]).lower() + ' '
+        s = s.rstrip()
+        s = set(s.split())
+        s = s.difference(stop_words)
+        m = set()
+        for token in s:
+            ids = s_inv_index.get(token, None)
+            if ids is not None:
+                m.update(ids)
+
+        # pick y/2 elements from m
+        k = min(y_pos, len(m))
+        m = list(m)
+        smpl_pos_neg = set()
+
+        while(len(smpl_pos_neg) < k):
+            num = random.choice(m)
+            smpl_pos_neg.add(num)
+            
+        # remaining y/2 items are selected here
+        while (len(smpl_pos_neg) < y):
+            rand_item_num = randint(0,s_tbl_sz-1)
+            smpl_pos_neg.add(rand_item_num)
+        h_table.update(smpl_pos_neg)
+
+    return h_table
+
+# down sample of two tables : based on sanjib's index based solution
+def down_sample(s_table, b_table, size, y):
+    """
+    This is down sample table function that down samples 2 tables A and B.
+
+    Args:
+        s_table: dataframe <input table A>
```

Comment:
why not have tableA and tableB as input variables?
Also for docstring arguments the format is,
variable_name (type): description
Eg., tableA (dataframe): input table A